src/constructors/kneser.c

```diff
@@ -0,0 +1,80 @@
+/*
+   IGraph library.
+   Copyright (C) 2022  The igraph development team
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+   02110-1301 USA
+*/
+
+#include "igraph_vector_list.h"
+
+static igraph_integer_t igraph_i_get_gcd(igraph_integer_t a, igraph_integer_t b){
+    while(b != 0) {
+        igraph_integer_t temp = a % b;
+        a = b;
+        b = temp;
+    }
+
+    return a;
+}
+
+static igraph_integer_t igraph_i_number_of_subsets(igraph_integer_t n, igraph_integer_t k){
+    igraph_integer_t numerator = 1, denominator = 1;
+    igraph_integer_t r = (k < n - k ? k : n - k);
+
+    while(r > 0) {
+        numerator *= n;
+        denominator *= r;
+        igraph_integer_t gcd = igraph_i_get_gcd(numerator, denominator);
+        numerator /= gcd;
+        denominator /= gcd;
+        --r;
+        --n;
+    }
+
+    return numerator;
+}
```

Comment:
I think there's a much easier approach that does not involve getting common denominators and does not risk overflow either. In pseudo-code:

```
result = 1
for i in 1..r:
    result = result * (n-i+1)
    result = result / i
```

Note that the order of operations is important; doing the multiplication and the division separately ensures that by the time you divide with `i` the result itself is already divisible by `i` due to previous multiplications.